search/includes/classes/class-search.php

```diff
@@ -229,6 +238,100 @@ public function do_mirror_search_request() {
 		] );
 	}
 
+	/**
+	 * Filter that runs at the end of WP_Query, used to transparently mirror certain WP_Query's on certain sites
+	 * for testing / evaluation
+	 */
+	public function filter__the_posts( $posts, $query ) {
+		// If this query is one that should be transparently mirrored, run the mirroring
+		$should_mirror = $this->should_mirror_wp_query( $query );
+
+		if ( $should_mirror ) {
+			$this->do_mirror_wp_query( $query );
+		}
+
+		return $posts;
+	}
+
+	public function should_mirror_wp_query( $query ) {
+		// If this was already an ES query, don't mirror
+		if ( isset( $query->query_vars['es'] ) || isset( $query->query_vars['ep_integrate'] ) || isset( $query->query_vars['vip_search_mirrored'] ) || isset( $query->elasticsearch_success ) ) {
+			return false;
+		}
+
+		// If mirroring is not enabled at all, skip
+		if ( ! $this->is_query_mirroring_enabled() ) {
+			return false;
+		}
+
+		// Is this one of the targeted queries?
+	
+		return false;
+	}
```

Comment:
It looks like you're missing some logic here.

Reply:
That's intentional - I want to get this shipped with no queries offloaded, then in subsequent PRs, start enabling it on a per-site/per-query basis using rules in here.

Reply:
Cool, sounds good.